tests/kafkatest/tests/client/client_compatibility_features_test.py

```diff
@@ -88,13 +88,7 @@ def invoke_compatibility_program(self, features):
         for k, v in features.iteritems():
             cmd = cmd + ("--%s %s " % (k, v))
         results_dir = TestContext.results_dir(self.test_context, 0)
-        try:
-            os.makedirs(results_dir)
-        except OSError as e:
-            if e.errno == errno.EEXIST and os.path.isdir(path):
-                pass
-            else:
-                raise
+        os.makedirs(results_dir)
```

Comment:
I don't think this substitution is correct.  `os.makedirs` "raises an error exception if the leaf directory already exists or cannot be created" which is not what we want here.  I do think it should be `os.path.isdir(results_dir)` rather than `os.path.isdir(path)`